File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {        
        vector<int>freq(201, 0);
        
        for(int i : nums)   
            ++freq[i];
        
        vector<vector<int>>res;
        
        while(true){
            vector<int>temp;            
            
            for(int i = 1; i <= 200; i++){
                if(freq[i] != 0){
                    temp.push_back(i);
                    --freq[i];
                }
            }
            
            if(temp.size() == 0)    break;
            
            res.push_back(temp);            
        }
        
        return res;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int freq[] = new int[201];
        
        for(int i : nums)
            ++freq[i];
        
        List<List<Integer>> res = new ArrayList<>();
        
        while(true){
            List<Integer> temp = new ArrayList<>();
            
            for(int i = 1; i < 201; i++){
                if(freq[i] != 0){
                    temp.add(i);
                    --freq[i];
                }
            }
            
            if(temp.isEmpty())  break;
            
            res.add(temp);
        }
        
        return res;
    }
}